src/polycap-source.c

```diff
@@ -664,72 +660,55 @@ polycap_transmission_efficiencies* polycap_source_get_transmission_efficiencies(
 		efficiencies->images->pc_exit_nrefl[j] = photon->i_refl;
 		efficiencies->images->pc_exit_dtravel[j] = photon->d_travel; //TODO: something is wrong with these values. Should be at least the length of the (poly)cap. Currently most values are less than the length...
 
-		#pragma omp critical
-		{
-		sum_irefl += photon->i_refl;
 		//Assign memory to arrays holding leak photon information (and fill them)
-		leak_mem_id[thread_id] = sum_nleaks;
-		recap_mem_id[thread_id] = sum_nrecap;
-		sum_nleaks += photon->n_leaks;
-		if(sum_nleaks > leak_mem_size){
+		n_leaks += photon->n_leaks;
+		if(n_leaks > leak_mem_size){
 			if (leak_mem_size == 0){
-				leak_mem_size = sum_nleaks;
+				leak_mem_size = n_leaks;
 			} else {
 				leak_mem_size *= 2;
-				if (leak_mem_size < sum_nleaks) leak_mem_size = sum_nleaks; //not doing this could be dangerous at low values
+				if (leak_mem_size < n_leaks) leak_mem_size = n_leaks; //not doing this could be dangerous at low values
 			}
-			efficiencies->images->leak_coords[0] = realloc(efficiencies->images->leak_coords[0], sizeof(double)* leak_mem_size);
-			efficiencies->images->leak_coords[1] = realloc(efficiencies->images->leak_coords[1], sizeof(double)* leak_mem_size);
-			efficiencies->images->leak_coords[2] = realloc(efficiencies->images->leak_coords[2], sizeof(double)* leak_mem_size);
-			efficiencies->images->leak_dir[0] = realloc(efficiencies->images->leak_dir[0], sizeof(double)* leak_mem_size);
-			efficiencies->images->leak_dir[1] = realloc(efficiencies->images->leak_dir[1], sizeof(double)* leak_mem_size);
-			efficiencies->images->leak_coord_weights = realloc(efficiencies->images->leak_coord_weights, sizeof(double)*n_energies* leak_mem_size);
-			efficiencies->images->leak_n_refl = realloc(efficiencies->images->leak_n_refl, sizeof(int64_t)* leak_mem_size);
+			leaks = realloc(leaks, sizeof(struct _polycap_leaks) * leak_mem_size);
```

Comment:
Nu faalt valgrind voornamelijk op deze lijn en op lijn 684.
Probleem doet zich, naar ik merk, enkel voor wanneer `n_leaks` = `leak_mem_size` (en ze dus allebei 0 zijn; analoog met `recap`). Nogal vreemd nu ik het schrijf, gezien ze dan normaal nooit in deze `if` mogen terecht komen indien dit het geval is...

Reply:
Net wat verbetering verkregen... Blijkbaar moet ik `leaks` en `recap` eerst `calloc`en. Ksnap niet goed waarom het relevant is dat de inhoud van die leak eerst op 0 wordt gezet, maar het helpt dus zo veel te beter :)

Reply:
Had je leaks en recap eerst op NULL gezet? `realloc` zal anders waarschijnlijk segfaulten anders. Die calloc zou idd niet nodig mogen zijn.

Reply:
hmm... Nee, niet gedaan. Zal aanpassen, want is eleganter dan die calloc :)